Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,130 @@
+            
+
+            // ~/.swiftpm/bin/
+            let swiftpmBinDir = dotSwiftPMDir.appending(component: "bin")
+            
+            if let path = env.path, !path.contains(swiftpmBinDir.pathString), env["SWIFTPM_SHUT_UP_ABOUT_EXECUTABLE_PATH"] == nil {
+                tool.observabilityScope.emit(warning: "PATH doesn't include \(swiftpmBinDir.pathString)! This means you won't be able to access the installed executables by default, and will need to specify the full path. If you wish to stop seeing this warning, set the enviroment variable `SWIFTPM_SHUT_UP_ABOUT_EXECUTABLE_PATH`.")
+            }
+            
+            let registryURL = dotSwiftPMDir.appending(component: "installedPackages.json").asURL
+            
+            var alreadyExisting = (try? InstalledPackage.registered(registryURL: registryURL)) ?? []
+            
+            let workspace = try tool.getActiveWorkspace()
+            let packageRoot = try tool.getPackageRoot()
+
+            let packageGraph = try tsc_await {
+                workspace.loadRootPackage(at: packageRoot, observabilityScope: tool.observabilityScope, completion: $0)
+            }
+            
+            let possibleCanidates = packageGraph.products.filter { $0.type == .executable }
+            let productToInstall: Product
+            
+            switch possibleCanidates.count {
+            case 0:
+                throw StringError("No Executable Products in Package.swift.")
+            case 1:
+                productToInstall = possibleCanidates[0]
+            default: // More than one, ask the user which one they wanna install
+                print("More than one executable target selected, please select of the following which you'd like to install by typing the number and pressing enter:")
+                for (index, product) in possibleCanidates.enumerated() {
+                    print("[\(index)] \(product.name)")
+                }
+                
+                guard let input = readLine(), let int = Int(input), possibleCanidates.indices.contains(int) else {
+                    throw StringError("Input should be a number between 0 and \(possibleCanidates.count - 1).")
+                }
+                
+                productToInstall = possibleCanidates[int]
+            }
+            
+            if let existingPkg = alreadyExisting.first(where: { $0.name == productToInstall.name }) {
+                throw StringError("\(productToInstall.name) is already installed at \(existingPkg.url)")
+            }
+            
+            try tool.createBuildSystem(explicitProduct: productToInstall.name).build(subset: .product(productToInstall.name))
+            
+            let binPath = try tool.buildParameters().buildPath.appending(component: productToInstall.name)
+            let finalBinPath = swiftpmBinDir.appending(component: binPath.basename)
+            try tool.fileSystem.move(from: binPath, to: finalBinPath)
```

Comment:
No need to remove from the original location, as the user may want to continue using the local product with `swift run`.

```suggestion
            try tool.fileSystem.copy(from: binPath, to: finalBinPath)
```